### DISCUSSION TITLE
Substitution for values from other sections with positional arguments

### DISCUSSION THREAD

**Discussion by 1Mark** [tox-dev/tox#2389] [General]
Lets say my config is 
```
[testenv]
{[testenv]test1}
{[testenv]test2}
    
[testenv:test1]
commands = pytest {posargs: -m blah1}

[testenv:test2]
commands = pytest -x {posargs: -m blah2}
```

I want to only allow posargs to be passed when running `tox -e test1` or `tox -e test2` but any posargs passed to tox directly shouldn't be forwarded.
That is, these work
```
tox -e test1 -- -m cats
tox -e test2 -- -m dogs
```

`tox -- -m cats` shouldn't error, or be ignored since I don't want pytest to be involved twice with the same args. 

Please note my main point is that I want the commands in `[testenv:testN]` to be reused inside `[testenv]` without allowing substitution AND not duplicating command in `[testenv:testN]`.
I.e. I don't think doing this is a good solution since if I alter `[testenv:test1]` I have to manually alter `[testenv]`

```
[testenv]
pytest {posargs: -m blah1}
pytest -x {posargs: -m blah2}

[testenv:test1]
commands = pytest {posargs: -m blah1}

[testenv:test2]
commands = pytest -x {posargs: -m blah2}
```

TL;DR I want to be able to  Substitute for values from other sections with positional arguments and not allow the top level command to forward the arguments. 

Maybe if I could store `[testenv:test1]`'s command in a variable then reuse that in `[testenv]` but it's unclear how to do this in the docs.

**Comment by gaborbernat** [comment]
> Please note my main point is that I want the commands in `[testenv:testN]` to be reused inside `[testenv]` without allowing substitution AND not duplicating command in `[testenv:testN]`.

We don't support this behaviour currently and I don't think we plan to support it. This would make the configuration be even more complicated than it is today and I'm personally not comfortable with it.

**Reply by 1Mark** [reply]
Do any workarounds exist?

**Reply by gaborbernat** [reply]
I don't think so. You'll just need to duplicate some settings :+1:
